rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py

```diff
@@ -0,0 +1,295 @@
+from abc import abstractmethod
+from typing import Any, Dict, Type
+
+from ray.rllib.algorithms.ppo import PPOConfig
+from ray.rllib.core import Columns
+from ray.rllib.core.models.base import ENCODER_OUT
+from ray.rllib.core.models.catalog import Catalog
+from ray.rllib.core.models.configs import MLPHeadConfig
+from ray.rllib.core.models.specs.specs_dict import SpecDict
+from ray.rllib.core.rl_module.rl_module import RLModule, SingleAgentRLModuleSpec
+from ray.rllib.core.rl_module.torch.torch_rl_module import TorchRLModule
+from ray.rllib.examples.envs.classes.correlated_actions_env import CorrelatedActionsEnv
+from ray.rllib.models.distributions import Distribution
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+from ray.rllib.utils.typing import TensorType
+
+from ray.tune import register_env
+
+register_env("correlated_actions_env", lambda _: CorrelatedActionsEnv(_))
+
+
+torch, nn = try_import_torch()
+
+
+class AutoregressiveActionRLM(RLModule):
+    def setup(self):
+        super().setup()
+
+        # Build the encoder.
+        self.encoder = self.config.get_catalog().build_encoder(framework=self.framework)
+
+        # Build the prior and posterior heads.
+        # Note, the action space is a Tuple space.
+        self.action_dist_cls = self.config.get_catalog().get_action_dist_cls(
+            self.framework
+        )
+        # Note further, we neet to know the required input dimensions for
+        # the partial distributions.
+        self.required_output_dims = self.action_dist_cls.required_input_dim(
+            space=self.config.action_space,
+            model_config=self.config.model_config_dict,
```

Comment:
Hmm, not sure I understand. In the TorchMultiDistribution, the `required_input_dim` takes an `input_lens` as 2nd arg, but here we pass in `module_config` as 2nd arg?

Reply:
These are named arguments. So nothing should happen, but good catch - because nothing happens with the second argument, too. Will erase it.